src/modules/card/card.service.ts

```diff
@@ -59,4 +76,131 @@ export class CardService {
+          createCardDTO.userProvisionalSolutionId,
+        );
+        if (!userProvisionalSolution)
+          throw new NotFoundCustomException(NotFoundCustomExceptionType.USER);
+        createCardDTO.userProvisionalSolutionName =
+          userProvisionalSolution.name;
+      }
+
+      if (createCardDTO.userAppProvisionalSolutionId) {
+        const userAppProvisionalSolution = await this.userService.findById(
+          createCardDTO.userAppProvisionalSolutionId,
+        );
+        if (!userAppProvisionalSolution)
+          throw new NotFoundCustomException(NotFoundCustomExceptionType.USER);
+        createCardDTO.userAppProvisionalSolutionName =
+          userAppProvisionalSolution.name;
+      }
+
+      if (createCardDTO.userDefinitiveSolutionId) {
+        const userDefinitiveSolution = await this.userService.findById(
+          createCardDTO.userDefinitiveSolutionId,
+        );
+        if (!userDefinitiveSolution)
+          throw new NotFoundCustomException(NotFoundCustomExceptionType.USER);
+        createCardDTO.userDefinitiveSolutionName = userDefinitiveSolution.name;
+      }
+
+      if (createCardDTO.userAppDefinitiveSolutionId) {
+        const userAppDefinitiveSolution = await this.userService.findById(
+          createCardDTO.userAppDefinitiveSolutionId,
+        );
+        if (!userAppDefinitiveSolution)
+          throw new NotFoundCustomException(NotFoundCustomExceptionType.USER);
+        createCardDTO.userAppDefinitiveSolutionName =
+          userAppDefinitiveSolution.name;
+      }
+
+      if (createCardDTO.managerId) {
+        const manager = await this.userService.findById(
+          createCardDTO.managerId,
+        );
+        if (!manager)
+          throw new NotFoundCustomException(NotFoundCustomExceptionType.USER);
+        createCardDTO.managerName = manager.name;
+      }
+
+      const lastInsertedCard = await this.cardRepository.find({order: {id: 'DESC'}, take: 1})
+      const {siteCardId} = lastInsertedCard[0]
+      createCardDTO.siteCardId = siteCardId+1
+      createCardDTO.siteCode = site.siteCode
+      createCardDTO.cardTypeColor = cardType.color
+      createCardDTO.areaName = area.name
+      createCardDTO.level = area.level
+      createCardDTO.superiorId = Number(area.superiorId) === 0 ? area.id: area.superiorId
+      createCardDTO.priorityCode = priority.priorityCode
+      createCardDTO.priorityDescription = priority.priorityDescription
+      createCardDTO.cardTypeMethodology = cardType.cardTypeMethodology
+      createCardDTO.cardTypeMethodologyName = cardType.methodology
+      createCardDTO.cardTypeName = cardType.name
+      createCardDTO.preclassifierCode = preclassifier.preclassifierCode
+      createCardDTO.preclassifierDescription = preclassifier.preclassifierDescription
+      createCardDTO.creatorName = creator.name
+      createCardDTO.responsableName = responsible.name
+      createCardDTO.createdAt = new Date()
+      createCardDTO.cardDueDate = new Date()
```

Comment:
we can remove this line since we're going to send it in another query

createCardDTO.cardDueDate

this one is going to be send front the client, not from he BE
 createCardDTO.createdAt

Reply:
@immanueldiaz 
ok but we have a litlle bit problem
"cardDueDate" is not null in database

Reply:
if is not null go ahead and keep it in the code

Reply:
and just to be sure
you want to send it from client "createdAt" (this one is related with "updatedAt" and "deletedAt" ) too?
is different to "cardCreationDate" (field that we mentioned in Jira)

Reply:
you;re right! sorry! my mistake ignore the comment and go ahead with this PR!

Reply:
dont worry